### DISCUSSION TITLE
I have no idea why c# 9 remove the Main() method??

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#6114] [Language Ideas]
I have no idea why c# 9 remove the Main() method??
all the static lang has main(), for: golang, java, C/c++,Delphi and   all the beginners  knows main() method. 
why are you remove main()?? 
this confused beginners .
this is unfriendly for beginners  to learning c#.

**Comment by svick** [comment]
It's not removed, it's just optional. If you want, you can continue to use (and teach) the `Main` method.

**Reply by ghost** [reply]
I know it`s not removed

I mean , create project use  visual studio.
![image](https://user-images.githubusercontent.com/15758574/167822792-4797e179-d293-4631-b312-2baf27030d1b.png)

**Reply by huoyaoyuan** [reply]
It's actually removed from the template in .NET 6, not in any C# version.
There is quick action to convert between the two forms, in VS 17.3:

<img width="469" alt="image" src="https://user-images.githubusercontent.com/5644458/168039102-ed1b2439-a374-4df8-add6-f59a75f302d2.png">

**Comment by TahirAhmadov** [comment]
If you want to create a project with Main method, use .NET 5 to create it, then edit csproj file to change to .NET 6.

**Comment by tannergooding** [comment]
> all the static lang has main(), for: golang, java, C/c++,Delphi and all the beginners knows main() method.

Yes, and many languages also have ways to avoid needing to explicitly have a `main` method

> why are you remove main()??

It's not removed, only optional now

> this confused beginners .
> this is unfriendly for beginners to learning c#.

This likely depends on who you ask and many beginners will actually find this simpler.

If you are learning a new programming language that has `Main` then one of the first examples you encounter is:
```csharp
using System;

public static class Program
{
    public static void Main(string[] args)
    {
        Console.WriteLine("Hello, World!");
    }
}
```

And almost inevitably the book or resource will state "we can ignore everything except this `Console.WriteLine()` for now". They do this because the "core" of learning a language isn't things like the classes or using statements or anything else. It's learning the basics about expressions so you can type something and get a result back. This is one of the reasons why languages like `python` and even `javascript` are so popular with beginners; they remove a lot of the "overhead" and allow you to jump straight into a REPL like environment.

This change is "different" for many existing developers, especially where they already have expectations of how programming works. However, it will likely make C# (and .NET) simpler and more approachable to those with no programming background and who are just getting started.

**Comment by siegfriedpammer** [comment]
You can also voice your opinion here: https://github.com/dotnet/docs/issues/27420

**Comment by zms9110750** [comment]
My current teaching route:
Variable - > if - > array - > method (actually local method) - > namespace - > class - > access modifier
I don't think we need to pay attention to access rights, static, namespace and methods. It will be more friendly to beginners.
The only problem is the lag of teaching materials.
The confusion of beginners comes from the inconsistency between the actual operation results and the teaching materials.

**Reply by zms9110750** [reply]
I think templates should be optional when creating projects.
They can't understand what the specific version is, so the template should be marked as "classic" and "new"
